src/consensus/params.h

```diff
@@ -39,20 +47,17 @@ struct BIP9Deployment {
 struct Params {
     uint256 hashGenesisBlock;
     int nSubsidyHalvingInterval;
-    /** Block height and hash at which BIP34 becomes active */
-    int BIP34Height;
+    /** Block hash at which BIP34 becomes active. Necessary to avoid checking BIP30. */
     uint256 BIP34Hash;
-    /** Block height at which BIP65 becomes active */
-    int BIP65Height;
-    /** Block height at which BIP66 becomes active */
-    int BIP66Height;
     /**
      * Minimum blocks including miner confirmation of the total of 2016 blocks in a retargeting period,
      * (nPowTargetTimespan / nPowTargetSpacing) which is also used for BIP9 deployments.
      * Examples: 1916 for 95%, 1512 for testchains.
      */
     uint32_t nRuleChangeActivationThreshold;
     uint32_t nMinerConfirmationWindow;
+    /** BIP90: Block height at which buried deployments becomes active */
+    uint64_t buried_deployments[MAX_BURIED_DEPLOYMENTS];
```

Comment:
Seems like it would fit in better with existing code to use `int` for block heights instead of `uint64_t`. Also would avoid signed/unsigned comparisons.